readme files for components and tests containing naming convention (#2897)
kyma-project/kyma: components/README.md

# Components

## Overview

The components folder contains the sources of all Kyma components.

> A Kyma component is any Pod/container/image deployed with and referenced in a Kyma module/chart to provide the module's functionality.

Author: An helm chart consists of a collection of kubernetes resource files/manifests. In that file, the docker images to use are specified by docker referencing the docker repository and image: gcr/kyma/event-bus:0.1.0

Each subfolder in the _components_ directory defines one component.

## Details

Every Kyma component has a dedicated folder containing its sources and a README.md containing further instructions on how to build and develop the component.

> The component name and with that the folder name should be suffixed with its component type. The types are defined as following:

Author: I vote for 2. option. However it is not only the folder name, it will be also the component name. So whenever we talk about the component later, we will use the folder name.

> The component name and with that the folder name should be suffixed with its component type. The types are defined as following:

Reviewer: ```suggestion
The component's name consists of a term describing the component, followed by a **component type**. The first part of the name may differ depending on the component's purpose. 
This table lists the available types:
```

| type|description|example|
|--|--|--|

> |controller|is a [Kubernetes Controller](https://kubernetes.io/docs/concepts/workloads/controllers/) reacting on standard Kubernetes resource or managing resources of a [Custom Resource Definition](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/). The name of the component reflects name of the primary resource which it controls.|namespace-controller|

Author: correct

> |operator|is a [Kubernetes Operator](https://coreos.com/operators/) covering  application specific logic for operatiion of the application, like steps to upscale a stateful applicatin. It usually reacts on changes to resources of a [Custom Resource Definition](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/). It uses the name of the application it operates. |application-operator|

Author: no, the operator knows how to operate an application. So when you want to upscale an app, the operator can do that task.
The sentence seems wrong now: 
which covers the application-specific logic the application uses to work, such a

> |operator|is a [Kubernetes Operator](https://coreos.com/operators/) covering  application specific logic for operatiion of the application, like steps to upscale a stateful applicatin. It usually reacts on changes to resources of a [Custom Resource Definition](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/). It uses the name of the application it operates. |application-operator|

Reviewer: I get it, I will rework the sentence to get the original meaning.

|job| A [Kubernetes Job](https://kubernetes.io/docs/tasks/job/) which performs a task once or periodically. It uses the name of the task it performs. |istio-patch-job (not renamed yet)|
|proxy| Acts as a proxy for an existing component, usually introducing a security model for this component. It uses the component's name. | apiserver-proxy|
|service| Serves an HTTP/S-based API, usually securely exposed to the public. It uses the domain name and the API it serves.|connector-service|
|broker| Implements the [OpenServiceBroker](https://www.openservicebrokerapi.org/) specification to enrich the Kyma Service Catalog with the services of a provider. It uses the name of the provider it integrates with.|azure-broker|

> |configurer| A one-time task usually executed as an [init container](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/) in order to configure an application.|ark-plugins-configurer (not migrated yet)|

Reviewer: ```suggestion
|configurer| A one-time task which usually runs as an [Init Container](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/) in order to configure the application.|ark-plugins-configurer (not migrated yet)|
```

# Components

## Overview

The `components` directory contains the sources of all Kyma components.
A Kyma component is any Pod, container, or image deployed with and referenced in a Kyma module or chart to provide the module's functionality.
Each subdirectory in the `components` directory defines one component.

## Details

> Every Kyma component resides in a dedicated folder, which contains its sources and a `README.md` file. This file provides instructions on how to build and develop the component.

Reviewer: ```suggestion
Every Kyma component resides in a dedicated folder which contains its sources and a `README.md` file. This file provides instructions on how to build and develop the component.
```

> The component's name consists of a term describing the component, followed by a **component type**. The first part of the name may differ depending on the component's purpose. 

Reviewer: ```suggestion
The component's name consists of a term describing the component, followed by the **component type**. The first part of the name may differ depending on the component's purpose. 
```
